Thoughtless-Eels/Bangazon-WebApp
# Create new product form

## Link to Ticket
Ticket: #1 

## Description of Proposed Changes

- Created a product model along with views for all CRUD methods
- added a link in the navbar to Sell A Product that only shows if user is logged in

## Steps to Test

Outline the steps to test

```sh
git fetch --all
git checkout gl-createProductForm
dotnet restore
```

Inside Visual Studio press the Run button in toolbar
Check that you see a "Sell A Product" link in navbar
If there, click the link and confirm that a page is loaded with a form to create a new product.
Logout of website and confirm that the "Sell A Product" link is not visible.



## Impacted Areas in Application

List general components of the application that this PR will affect:

* ProductController
* Views for Product CRUD methods (create, delete etc)
* _SellProductPartial.cshtml 


## Mentions @username

Tag users that need to review this code

## Definition of Done

1. The project must be fully documented. This includes the following:
    1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
    1. Every class must be documented with purpose, author, and methods.
    1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each.
